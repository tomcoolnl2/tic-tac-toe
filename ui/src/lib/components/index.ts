export { AppLogo } from './logo/logo';
export { Avatar } from './avatar/avatar';
export { Button } from './button/button';
export { Cell } from './cell/cell';
export { ChooseDifficulty } from './choose-difficulty/choose-difficulty';
export { GameOver } from './game-over/game-over';
export { Input } from './input/input';
export { Modal } from './modal/modal';
export { ReloadButton } from './reload-button/reload-button';
export { ScoreBoardItem } from './score-board-item/score-board-item';
export { Spinner } from './spinner/spinner';
export { SymbolChoice } from './symbol-choice/symbol-choice';
export { TurnIndicator } from './turn-indicator/turn-indicator';
