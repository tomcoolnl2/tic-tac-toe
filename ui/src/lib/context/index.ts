export * from './content/content';
export * from './auth/auth';
