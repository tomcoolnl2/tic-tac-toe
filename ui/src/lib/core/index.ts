export * from './canvas';
export * from './grid';
export * from './flex-box';
export * from './divider';
