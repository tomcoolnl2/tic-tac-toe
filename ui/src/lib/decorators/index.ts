export { withTheme } from './withTheme';
export type { ContainerProps } from './withContainer';
export { withContainer } from './withContainer';
