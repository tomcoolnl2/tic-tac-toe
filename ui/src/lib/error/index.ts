export * from './error-boundary';
export { AuthError } from '../context/auth/error';
