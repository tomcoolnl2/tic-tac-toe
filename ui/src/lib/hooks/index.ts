export * from './useAnimationFrame';
export * from './useBehaviorSubjectState';
export * from './useScreenOrientation';
export * from './useSettingsHandlers';
export * from './useInterfaceHandlers';
export * from './useTimer';
