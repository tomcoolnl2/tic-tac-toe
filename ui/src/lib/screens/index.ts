export { LoadingScreen } from './loading';
export { LoginScreen } from './login';
export { ErrorScreen } from './error';
export { GameScreen } from './game';
export { SettingsScreen } from './settings';
export { RestartModalScreen } from './reload';
export { GameOverModalScreen } from './game-over';
