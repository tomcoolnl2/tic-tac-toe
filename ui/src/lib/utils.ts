/**
 * Retrieves the HTMLButtonElement from a MouseEvent's target.
 * @param {MouseEvent} event - The mouse event.
 * @returns {HTMLButtonElement} The HTMLButtonElement extracted from the event target.
 */
export const getEventTargetElement = (event: MouseEvent): HTMLButtonElement =>
	event.target as HTMLButtonElement;

/**
 * Returns the value of a specified data attribute from an HTMLButtonElement.
 * @param {string} attribute - The data attribute to retrieve.
 * @returns {(element: HTMLButtonElement) => string | undefined} A function that accepts an HTMLButtonElement and returns the specified data attribute value.
 */
export const getDataSetAttribute =
	(attribute: string): ((element: HTMLButtonElement) => string | undefined) =>
	(element: HTMLButtonElement) =>
		element.dataset[attribute];

/**
 * Asynchronously pauses the execution for a specified duration.
 * @param {number} ms - The duration in milliseconds to wait before resolving the promise.
 * @returns {Promise<void>} A promise that resolves after the specified duration.
 */
export const sleep = (ms: number): Promise<void> => new Promise((r) => setTimeout(r, ms));
