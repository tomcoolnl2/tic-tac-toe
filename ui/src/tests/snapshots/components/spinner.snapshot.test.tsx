import { render } from '@testing-library/react';
import { Spinner } from '../../../lib/components';

test('Spinner component snapshot test', () => {
	const { container } = render(<Spinner />);
	expect(container).toMatchSnapshot();
});
