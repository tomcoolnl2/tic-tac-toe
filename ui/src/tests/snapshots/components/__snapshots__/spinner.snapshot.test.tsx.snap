// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component snapshot test 1`] = `
<div>
  <div
    class="spinner"
  />
</div>
`;
